server.go

```diff
@@ -126,14 +127,46 @@ func (server *Server) Stop() {
 		server.Log.Infof("waiting for requests to finish")
 		server.listener.Wait()
 	}
+
+	server.waitForConnectionsToClose()
+}
+
+// StopDeadline is the hard deadline, after which Server.Stop() does no longer
+// wait for connections to close and returns.
+var StopDeadline = time.Minute
+
+// waitForConnectionsToClose waits until all the handlers from listener and
+// tlsListener are done and closed their connections.
+func (server *Server) waitForConnectionsToClose() {
+	deadline := time.NewTimer(StopDeadline)
+	for {
+		if server.numConns.Count() == 0 {
+			server.Log.Infof("all requests finished")
+			break
+		}
+
+		select {
+		case <-deadline.C:
+			server.Log.Warnf("not all requests finished after %v", StopDeadline)
+			deadline.Stop()
```

Comment:
"Stop prevents the Timer from firing" - why do u need it _after_ it fired already?

Reply:
true, I remove that